### PR TITLE
Multiple rms closest

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [ ] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Added feature which does something
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Fixed bug identified in (`#YY <https://github.com/GranthamImperial/silicone/issues/YY>`_)
```

### CODE REVIEW

**Comment by znicholls** [comment]
@Rlamboll still a draft but wanted to get your thoughts at this stage before I write an example notebook

**Comment by Rlamboll** [comment]
Hi Zeb, I like the idea of the tool and the syntax for infill multiple seems fine. I find the interrupting loops in the code quite hard to follow, so more comments would be useful, as well as more documentation since it's unusual for a cruncher to have this sort of ability. I'd also do more tests, like a test where it's possible to return the wrong answer for two of the variables without inventing new data - currently you are choosing the right pathway from only one option on the N2O two-pathway use case!

**Comment by znicholls** [comment]
> interrupting loops

Is this a synonym for nested loops?

> like a test where it's possible to return the wrong answer for two of the variables without inventing new data

Ok cool will have a think. If you have any more you want to suggest please fire away (if you want to make a PR into this one with the use cases of interest that could also work well).

**Comment by Rlamboll** [comment]
No, by interrupting I mean loops featuring continues and breaks. I don't belong to the hard school of "never use these, put it all in an if statement or control your loops" but a few comments would go a long way in increasing readibility. 

I'm quite busy now and writing tests for this would be low on my to-do list but I can do it in the distant future.

**Comment by znicholls** [comment]
> No, by interrupting I mean loops featuring continues and breaks. I don't belong to the hard school of "never use these, put it all in an if statement or control your loops" but a few comments would go a long way in increasing readibility.

Ye cool got it

> I'm quite busy now and writing tests for this would be low on my to-do list but I can do it in the distant future.

Yep let's leave this sitting as a proof of concept until we find time again
